hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdiff/RocksDBCheckpointDiffer.java

```diff
@@ -471,38 +539,213 @@ public HashSet<String> readRocksDBLiveFiles(String dbPathArg) {
     return liveFiles;
   }
 
-  // Given the src and destination Snapshots, it prints a Diff list.
-  private synchronized void printSnapdiffSSTFiles(
-      Snapshot src, Snapshot dest) throws RocksDBException {
-    LOG.warn("Src Snapshot files :" + src.dbPath);
+  private String[] sstTokensRead;
+  private long reconstructionSnapshotGeneration;
+
+  /**
+   * Process each line of compaction log text file input and populate the DAG.
+   */
+  private synchronized void processCompactionLogLine(String line) {
+
+    LOG.debug("Processing line: {}", line);
+
+    // Skip comments
+    if (line.startsWith("#")) {
+      LOG.debug("Comment line, skipped");
+      return;
+    }
+
+    // Read sequence number as the snapshot generation for the nodes
+    if (line.startsWith(COMPACTION_LOG_SEQNUM_LINE_PREFIX)) {
+      LOG.debug("Reading sequence number as snapshot generation");
+      final String seqNumStr =
+          line.substring(COMPACTION_LOG_SEQNUM_LINE_PREFIX.length()).trim();
+      reconstructionSnapshotGeneration = Long.parseLong(seqNumStr);
+      return;
+    }
+
+    if (sstTokensRead == null) {
+      // Store the tokens in the first line
+      sstTokensRead = line.split("\t");
+      LOG.debug("Number of input files: {}", sstTokensRead.length);
+      if (sstTokensRead.length == 0) {
+        // Sanity check. inputFiles should never be empty. outputFiles could.
+        throw new RuntimeException(
+            "Compaction inputFiles list is empty. File is corrupted?");
+      }
+    } else {
+      final String[] outputFilesRead = line.split("\t");
+      LOG.debug("Number of output files: {}", outputFilesRead.length);
+
+      // Populate the compaction DAG
+      populateCompactionDAG(sstTokensRead, outputFilesRead,
+          reconstructionSnapshotGeneration);
+
+      // Reset inputFilesRead to null so
+      sstTokensRead = null;
+    }
+  }
+
+  /**
+   * Helper to read compaction log to the internal DAG.
+   * <p>
+   * DO NOT use this function in another context without understanding what it
+   * does, due to the state preserved between calls (in sstTokensRead).
+   */
+  private void readCompactionLogToDAG(String currCompactionLogPath) {
```

Comment:
We should also trigger it on OM restart

Reply:
Done. Now previous compaction logs will **only** be loaded during OM startup.